index.d.ts

```diff
@@ -1,12 +1,65 @@
-import { Parser } from 'acorn' 
+import * as acorn from 'acorn';
 
-declare const jsx: (options?: jsx.Options) => (BaseParser: typeof Parser) => typeof Parser;
+declare const jsx: {
+  tokTypes: typeof acorn.tokTypes;
```

Comment:
Regarding use of named exports, [Node.js docs](https://nodejs.org/api/esm.html) state:

> CommonJS modules consist of a module.exports object...
> <SNIP>
> The detection of named exports is based on common syntax patterns but does not always correctly detect named exports. In these cases, using the default import form described above can be a better option.

So I'm wondering if that would be safe.

However, you are right that I should be using the jsx tokTypes instead of the Acorn one here. But it seems then that trying to fix this brings up two problems:

1. Errors are given suggesting that the `jsx` in this case will be understood as the exported `const` rather than the namespace which we want. Renaming the namespace to `jsxNS`, fixes the error but then the namespace isn't exported.
2. And because only one export is allowed at the top level, I can't apparently move `interface TokTypes` to the top level without being redundant (i.e., I can have one copy at the top level which is not exported and one copy within the namespace which, as you say, is auto-exported).

Reply:
I solved this I think by reexporting the non-exported interface with an exported type alias. Let me know if you think this is all suitable.